### ISSUE TITLE
Add support for salesforce grpc pubsub api

### ISSUE THREAD

**Issue by jsight** [apache/camel-kamelets#1546]
Camel 4.x will have support for the grpc pubsub api within salesforce. Will kamelets add this?

Also, does the kamelet salesforce connector support keystores for authentication? I didn't see the parameters.

**Comment by oscerd** [comment]
You're welcome to work on the feature.

**Comment by oscerd** [comment]
https://camel.apache.org/components/next/salesforce-component.html#PubSubAPI

For the keystores etc we could add that.

**Comment by jsight** [comment]
I did think about it, but I'm not sure how helpful it would be. I don't think we have a way to store and maintain the replayId and that is essential for a real application of this.

**Comment by oscerd** [comment]
Why there is no way of maintaining the replayId?

**Comment by jsight** [comment]
It would make a huge difference in usability if persistence were possible, especially across clusters.

**Comment by oscerd** [comment]
Two questions here, and they have different answers.

## "Will kamelets add gRPC pub/sub support?"

Yes — #3012 adds `salesforce-pubsub-source`, which subscribes through the gRPC Pub/Sub API rather than the older streaming API that `salesforce-source` uses.

The component side has been ready for a while; the catalog simply had not caught up. `camel-salesforce` in 4.22 carries both operations:

```
operationName enum -> [..., 'subscribe', 'pubSubSubscribe', 'pubSubPublish']
```

That PR covers `pubSubSubscribe`. A `pubSubPublish` sink is the obvious companion and is a clean follow-up.

This overlaps #2263, which asks for the same thing. I have pointed the PR at that one and left this open, because your second question is separate and still unanswered.

## "Does the kamelet salesforce connector support keystores for authentication?"

**No, and you were right that the parameters are not there.** Every Salesforce Kamelet in the catalog builds its component bean with username/password only:

```yaml
      - name: local-salesforce
        type: "#class:org.apache.camel.component.salesforce.SalesforceComponent"
        properties:
          clientId: "{{clientId}}"
          clientSecret: "{{clientSecret}}"
          userName: "{{userName}}"
          password: "{{password}}"
          loginUrl: "{{loginUrl}}"
```

The component itself does support JWT bearer flow — `authenticationType`, `jwtAudience` and `keystore` are all component options — so the gap is in the Kamelets, not in Camel.

The awkward part is `keystore`: it is typed `org.apache.camel.support.jsse.KeyStoreParameters`, an object rather than a scalar. So exposing JWT auth means either building a `KeyStoreParameters` bean inside each template from scalar properties, or accepting a bean reference from the operator. That is the same shape as the decision in #3000 for the OpenSearch Kamelets, and it would need applying consistently across all five Salesforce Kamelets.

That is a real piece of work with a design choice in it rather than something to fold into the Pub/Sub PR, so I have not attempted it here. If you want it tracked properly it deserves its own issue — happy to write it up with the options laid out.

---
_Claude Code on behalf of Andrea Cosentino_

**Comment by oscerd** [comment]
Split the authentication half out into #3014 so it survives this issue being closed once the Pub/Sub work lands.

That issue records the state across all five existing Salesforce Kamelets (username/password only), what the component actually offers (`authenticationType`, `jwtAudience`, `keystore`, `instanceUrl`), and the one decision that is not mechanical — `keystore` is typed `KeyStoreParameters`, so it is either built in-template from scalars or accepted as a bean reference.

---
_Claude Code on behalf of Andrea Cosentino_

**Comment by oscerd** [comment]
This issue asked two separate things, and they have diverged — one is delivered, the other is tracked elsewhere.

**gRPC Pub/Sub support: done.** `salesforce-pubsub-source` is in the catalog, added in 13f59f7d9 (#3012, for #2263). It consumes the Salesforce Pub/Sub API over gRPC, with `replayPreset` / `replayId` for positioning, `batchSize`, and `deserializeType` for choosing between the raw Avro payload and a POJO.

**Keystore authentication: still accurate, and now tracked in its own issue.** The second half of this report — *"does the kamelet salesforce connector support keystores for authentication? I didn't see the parameters"* — is still true. The new Kamelet declares `clientId`, `clientSecret`, `userName` and `password`, so it is username/password only, and the other Salesforce Kamelets are the same. `camel-salesforce` itself supports JWT bearer with a keystore, so this is a gap in the Kamelets rather than in the component.

That gap is #3014, which covers all the Salesforce Kamelets rather than only the Pub/Sub one, so it is the better place for it.

Closing this one: the feature it asked for exists, and the authentication question is live in #3014.

---
_Claude Code on behalf of Andrea Cosentino_
